source/common/upstream/subset_lb.cc

```diff
@@ -118,6 +177,17 @@ HostConstSharedPtr SubsetLoadBalancer::chooseHost(LoadBalancerContext* context)
       // Subset lookup succeeded, return this result even if it's nullptr.
       return host;
     }
+    // otherwise check if there is fallback policy configured for given route metadata
+    absl::optional<
+        envoy::api::v2::Cluster::LbSubsetConfig::LbSubsetSelector::LbSubsetSelectorFallbackPolicy>
+        selector_fallback_policy = tryFindSelectorFallbackPolicy(context);
+    if (selector_fallback_policy) {
+      // return result according to configured fallback policy
+      HostConstSharedPtr host =
+          chooseHostForSelectorFallbackPolicy(selector_fallback_policy, context);
+      // Subset lookup succeeded, return this result even if it's nullptr.
+      return host;
```

Comment:
nit: elide the variable and just `return chooseHostForSelectorFallbackPolicy(...)`